src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -43,27 +47,37 @@ public class ScriptHostManager : IScriptHostEnvironment, IDisposable
         private ILogger _logger;
 
         private ScriptSettingsManager _settingsManager;
-        private AutoRecoveringFileSystemWatcher _scriptFileWatcher;
         private CancellationTokenSource _restartDelayTokenSource;
 
-        public ScriptHostManager(ScriptHostConfiguration config, IScriptHostEnvironment environment = null)
-            : this(config, ScriptSettingsManager.Instance, new ScriptHostFactory(), environment)
+        public ScriptHostManager(ScriptHostConfiguration config, IScriptEventManager eventManager = null, IScriptHostEnvironment environment = null)
+            : this(config, ScriptSettingsManager.Instance, new ScriptHostFactory(), eventManager, environment)
         {
             if (config.FileWatchingEnabled)
             {
-                _scriptFileWatcher = new AutoRecoveringFileSystemWatcher(config.RootScriptPath);
-                _scriptFileWatcher.Changed += OnScriptFileChanged;
+                _fileEventSource = new FileWatcherEventSource(EventManager, "ScriptFiles", config.RootScriptPath);
```

Comment:
It looks like ScriptHost sets up the same `FileWatcherEventSource`. Does that mean they're both registering two watchers of the same files? Would that cause dupe events to fire?